app/lib/features/chat/widgets/room_avatar.dart

```diff
@@ -130,37 +165,63 @@ class RoomAvatar extends ConsumerWidget {
     );
   }
 
-  Widget groupDMAvatarUI(List<Member> members, WidgetRef ref) {
-    return Stack(
-      alignment: Alignment.bottomLeft,
-      clipBehavior: Clip.none,
-      children: [
-        memberAvatar(members[0], ref),
-        Positioned(
-          left: -7,
-          bottom: -5,
-          child: memberAvatar(members[1], ref),
-        ),
-        Positioned.fill(
-          bottom: -5,
-          child: Align(
-            alignment: Alignment.bottomRight,
-            child: Container(
-              width: 15,
-              height: 15,
-              alignment: Alignment.center,
-              decoration: const BoxDecoration(
-                color: Colors.green,
-                shape: BoxShape.circle,
+  Widget groupAvatarDM(List<Member> members, WidgetRef ref) {
+    final userId = members[0].userId().toString();
+    final secondaryUserId = members[1].userId().toString();
+    final profile = ref.watch(memberProfileProvider(members[0]));
+    final secondaryProfile = ref.watch(memberProfileProvider(members[1]));
+
+    return profile.when(
+      data: (data) {
+        return ActerAvatar(
+          avatarInfo: AvatarInfo(
+            uniqueId: userId,
+            displayName: data.displayName,
+            avatar: data.getAvatarImage(),
+          ),
+          avatarsInfo: secondaryProfile.maybeWhen(
+            data: (secData) => [
+              AvatarInfo(
+                uniqueId: secondaryUserId,
+                displayName: secData.displayName,
+                avatar: secData.getAvatarImage(),
               ),
-              child: Text(
-                '+${members.length - 2}',
-                style: const TextStyle(fontSize: 8),
+              for (int i = 2; i < members.length; i++)
+                AvatarInfo(
+                  uniqueId: members[i].userId().toString(),
+                ),
+            ],
+            orElse: () => [],
```

Comment:
shouldn't we fallback to just using all userIds?

```suggestion
            orElse: () => [
              for (int i = 1; i < members.length; i++)
                AvatarInfo(
                  uniqueId: members[i].userId().toString(),
                ),
              ]
```

Reply:
Yeah you're correct. Thanks for noticing it out.